python/lib/client/dmod/client/request_clients.py

```diff
@@ -156,14 +364,141 @@ def _update_after_valid_response(self, response: MaaSDatasetManagementResponse):
         # TODO: think about if anything is needed for this
         pass
 
-    async def create_dataset(self, name: str, category: DataCategory) -> bool:
+    async def _upload_file(self, dataset_name: str, path: Path, item_name: str) -> bool:
+        """
+        Upload a single file to the dataset
+
+        Parameters
+        ----------
+        dataset_name : str
+            The name of the destination dataset.
+        path : Path
+            The path of the local file to upload.
+        item_name : str
+            The name of the destination dataset item in which to place the data.
+        Returns
+        -------
+        bool
+            Whether the data upload was successful.
+        """
+        await self._async_acquire_session_info()
+        #raw_data = path.read_bytes()
+        chunk_size = 1024
+        message = MaaSDatasetManagementMessage(action=ManagementAction.ADD_DATA, dataset_name=dataset_name,
+                                               session_secret=self.session_secret, data_location=item_name)
+        async with websockets.connect(self.endpoint_uri, ssl=self.client_ssl_context) as websocket:
+            with path.open() as file:
+                raw_chunk = file.read(chunk_size)
+                while True:
```

Comment:
A little more complexity, and we have application layer TCP 😆 .  Might be worth considering some of those features, though, in all seriousness.  Perhaps some message retries before total failure?

Reply:
Yeah, at the time I was in a hurry to shoehorn in a solution, and did not come up with the optimal one.  This will need to be revisited, which is already on my radar.